lib/Push.php

```diff
@@ -246,6 +246,8 @@ public function pushToDevice(int $id, INotification $notification, ?OutputInterf
 			$language = $this->l10nFactory->getUserLanguage($user);
 			$this->printInfo('Language is set to ' . $language);
 
+			$this->notificationManager->preloadMany([$notification], $language);
```

Comment:
During pushing it sounds a bit weird, but there apps can parse the notification already before sending it. And that should be done, at least grouped by language, so we don't do the same preparsing multiple times.
We do that in Talk already since some time. Documentation is pending and will be done as perparation for my Lightning Talk that I submitted for the Conf.

- Ref https://github.com/nextcloud/notifications/pull/1753
- Ref https://github.com/nextcloud/spreed/pull/12303

In case of a push also the information should be there already as the app just created the notification object with the details 🙈 

But I guess structurally it makes sense to always have the same calling order.

Reply:
Yeah, because I was thinking about consistency. Can consumers/apps expect that the method is always called? Otherwise, they would have to implement the data fetching logic twice, once in `preloadMany` and once in `prepare`.